staff/abdou-bousiyd/precourse/airlines-pro/index.js

```diff
@@ -0,0 +1,131 @@
+
+const messages = {
+    
+    hiAdmin: `Hi Admin\n you can add flights or remove them\n write an action ADD or DELETE`,
+    deleteFlightConfirm: "Do you want to delete some flight?",
+    addMoreConfirm: "Do you want to add more flights?",
+    addSuccess: "flight added",
+    ActionLimited: "you have exceeded the flight limit",
+
+    cancelled: "You pressed Cancel",
+    userActions: "You can see flights by price",
+    selectAFlyConfirm: "Do you want to buy a flight?",
+    selectAFlyAlert: "Thanks for your purchase",
+    userAlert: "Thanks for your visit",
+    selectAFlyPrompt: "Enter ID",
+    selectAFlyPrint: "You have not made any purchases."
+}
+
+class AirlinesPro {
+    constructor() {
+        this.flights = [
+            { id: 0, to: 'Bilbao', from: 'Barcelona', cost: 1600, scale: false },
+            { id: 1, to: 'New York', from: 'Barcelona', cost: 700, scale: false },
+            { id: 2, to: 'Los Angeles', from: 'Madrid', cost: 1100, scale: true },
+            { id: 3, to: 'Paris', from: 'Barcelona', cost: 210, scale: false },
+            { id: 4, to: 'Roma', from: 'Barcelona', cost: 150, scale: false },
+            { id: 5, to: 'London', from: 'Madrid', cost: 200, scale: false },
+            { id: 6, to: 'Madrid', from: 'Barcelona', cost: 90, scale: false },
+            { id: 7, to: 'Tokyo', from: 'Madrid', cost: 1500, scale: true },
+            { id: 8, to: 'Shangai', from: 'Barcelona', cost: 800, scale: true },
+            { id: 9, to: 'Sydney', from: 'Barcelona', cost: 150, scale: true },
+            { id: 10, to: 'Tel-Aviv', from: 'Madrid', cost: 150, scale: false }
+        ];
+        this.role = 'USER'
+    }
+
+    print(messages) {
+        console.log(messages)
+    }
+
+    loginUser() {
+        let user = prompt("ADMIN / USER").toUpperCase()
+        return (!user) ? this.loginUser() : this.role = user
+    }
+
+    ejecutarActions() {
+        if (this.role === 'USER') this.userActions();
+        if (this.role === 'ADMIN') this.adminActions()
+    }
+
+    adminActions() {
+        let action = prompt(messages.hiAdmin).toUpperCase()
```

Comment:
It might not be a very good idea chaining `.toUpperCase()` with the prompt function because if the user cancels it, it'll return null and this will show up in console as `can't set upper case on null value`. Something like that